src/controllers/BaseController.js

```diff
@@ -3,35 +3,55 @@ class BaseController {
     this.service = service;
   }
 
-  async getAllData(req, res) {
-    const result = await this.service.getAllData(req.query);
-    res.status(200).json(result);
+  async getAllData(req, res, next) {
+    const promise = this.service.getAllData(req.query);
+    const result = await BaseController.checkError(promise, next);
+    res.status(200)
+      .json(result);
   }
 
-  async getById(req, res) {
-    const result = await this.service.getById(req.params.id);
-    res.status(200).json(result);
+  async getById(req, res, next) {
+    const promise = this.service.getById(req.params.id);
+    const result = await BaseController.checkError(promise, next);
+    res.status(200)
+      .json(result);
   }
 
-  async insertData(req, res) {
-    const result = await this.service.insertData(req.body);
-    res.status(200).json(result);
+  async insertData(req, res, next) {
+    const promise = this.service.insertData(req.body);
+    const result = await BaseController.checkError(promise, next);
+    res.status(200)
+      .json(result);
   }
 
-  async deleteById(req, res) {
-    const result = await this.service.deleteById(req.params.id);
+  async deleteById(req, res, next) {
+    const promise = this.service.deleteById(req.params.id);
+    const result = await BaseController.checkError(promise, next);
     const row = result.affected;
     if (row) {
-      res.status(202).json({
-        id: req.params.id,
-        message: 'data has been deleted',
-      });
+      res.status(202)
+        .json({
+          id: req.params.id,
+          message: 'data has been deleted',
+        });
     }
   }
 
-  async updateById(req, res) {
-    const result = await this.service.updateById(req.params.id, req.body);
-    res.status(200).json(result.raw[0]);
+  async updateById(req, res, next) {
+    const promise = this.service.updateById(req.params.id, req.body);
+    const result = await BaseController.checkError(promise, next);
+    res.status(200)
+      .json(result.raw[0]);
+  }
+
+  static async checkError(promise, next) {
```

Comment:
Avoid copy past
This way is cleaner
https://appdividend.com/2018/09/16/express-error-handling-example-tutorial/

Reply:
> Avoid copy past
> This way is cleaner
> https://appdividend.com/2018/09/16/express-error-handling-example-tutorial/

Did you mean to do it like this:
router.get ('/', wrapAsync (userController.getUsers.bind (userController))) ;? 
Should this wrapping function be a controller method?